openmc/universe.py

```diff
@@ -600,3 +627,112 @@ def _determine_paths(self, path='', instances_only=False):
             cell._num_instances += 1
             if not instances_only:
                 cell._paths.append(cell_path)
+
+
+class DAGMCUniverse(UniverseBase):
+    """A reference to a DAGMC file to be used in the model.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the DAGMC file used to represent this universe.
+    universe_id : int, optional
+        Unique identifier of the universe. If not specified, an identifier will
+        automatically be assigned.
+    name : str, optional
+        Name of the universe. If not specified, the name is the empty string.
+    auto_geom_ids : bool
+        Set IDs automatically on initialization (True) or report overlaps
+        in ID space between CSG and DAGMC (False)
+    auto_mat_ids : bool
+        Set IDs automatically on initialization (True)  or report overlaps
+        in ID space between OpenMC and UWUW materials (False)
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier of the universe
+    name : str
+        Name of the universe
+    filename : str
+        Path to the DAGMC file used to represent this universe.
+    auto_geom_ids : bool
+        Set IDs automatically on initialization (True) or report overlaps
+        in ID space between CSG and DAGMC (False)
+    auto_mat_ids : bool
+        Set IDs automatically on initialization (True)  or report overlaps
+        in ID space between OpenMC and UWUW materials (False)
+    """
+
+    def __init__(self,
+                 filename,
+                 universe_id=None,
+                 name='',
+                 auto_geom_ids=False,
+                 auto_mat_ids=False):
+        super().__init__(universe_id, name)
+        # Initialize class attributes
+        self.filename = filename
+        self.auto_geom_ids = auto_geom_ids
+        self.auto_mat_ids = auto_mat_ids
+
+    def __repr__(self):
+        fmt_str = '{: <16}=\t{}\n'
+        string = super().__repr__()
+        string += fmt_str.format('\tGeom', 'DAGMC')
+        string += fmt_str.format('\tFile', self.filename)
+        return string
+
+    @property
+    def filename(self):
+        return self._filename
+
+    @filename.setter
+    def filename(self, val):
+        cv.check_type('DAGMC filename', val, str)
+        self._filename = val
+
+    @property
+    def auto_geom_ids(self):
+        return self._auto_geom_ids
+
+    @auto_geom_ids.setter
+    def auto_geom_ids(self, val):
+        cv.check_type('DAGMC automatic geometry ids', val, bool)
+        self._auto_geom_ids = val
+
+    @property
+    def auto_mat_ids(self):
+        return self._auto_mat_ids
+
+    @auto_mat_ids.setter
+    def auto_mat_ids(self, val):
+        cv.check_type('DAGMC automatic material ids', val, bool)
+        self._auto_mat_ids = val
+
+    def clone(self, clone_materials=True, clone_regions=True, memo=None):
+        pass
```

Comment:
Should this be a warning or something?

Reply:
I need to think about this a bit more. I don't see any reason why we can't clone a `DAGMCUniverse`. I wouldn't necessarily recommend it, but it should be possible.